python/pylibvw.cc

```diff
@@ -700,38 +784,38 @@ void unsetup_example(vw_ptr vwP, example_ptr ae)
   ae->num_features = 0;
   ae->reset_total_sum_feat_sq();
   ae->loss = 0.;
-
   if (all.ignore_some) { THROW("error: cannot unsetup example when some namespaces are ignored!"); }
 
   if (all.skip_gram_transformer != nullptr && !all.skip_gram_transformer->get_initial_ngram_definitions().empty())
   { THROW("error: cannot unsetup example when ngrams are in use!"); }
 
   if (all.add_constant)
   {
-    ae->feature_space[constant_namespace].clear();
-    int hit_constant = -1;
+    (*ae->feature_space.namespace_index_begin(constant_namespace)).clear();
```

Comment:
The constant namespace should be removed not just cleared

Reply:
The constant namespace is removed at the end of the function, along with any namespaces which are after it